zip_0244.py

```diff
@@ -357,19 +400,20 @@ def randbytes(l):
             ('tx', {'rust_type': 'Vec<u8>', 'bitcoin_flavoured': False}),
             ('txid', '[u8; 32]'),
             ('auth_digest', '[u8; 32]'),
+            ('amounts', '[i64; %d]' % len(t_inputs)),
+            ('script_codes', {
+                'rust_type': '[Vec<u8>; %d]' % len(t_inputs),
```

Comment:
Kris noted that this causes a bug in the output, because `len(t_inputs)` returns the length for the _last_ test vector, but it varies across them. This should be:
```suggestion
            ('amounts', 'Vec<i64>'),
            ('script_codes', {
                'rust_type': 'Vec<Vec<u8>>',
```
but that won't work without more changes to the Rust output renderer, so instead I'm going to overhaul the latter after this PR.